# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Abdelrahmanshaheen1/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

### Lane 2 — Refresh / Content Opportunity Scoring

I chose the Refresh / Content Opportunity Scoring lane. My project will investigate which website pages should be reviewed first for actions such as refreshing, expanding, protecting, pruning, or monitoring. I chose this lane because content teams may have thousands of pages but limited time, so they need a useful way to prioritize their work. The starter data contains page-level signals such as impressions, clicks, average position, CTR, content age, freshness, engagement, and recent trend information. These signals may help create a ranked review queue with clear reasons for each recommendation. The starter notebooks also showed that combining several signals can produce a more useful ranking than relying only on one fixed rule, making this lane worth investigating further.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Research question

Which website pages should a content team review first because they show signs of decline or a useful opportunity for improvement?

- **Unit of analysis:** One webpage.
- **Output:** A ranked list of pages, with a priority score and readable reasons for each recommendation.
- **Decision improved:** Which pages should the content team spend its limited review time on first?
- **Who acts:** A content editor, SEO specialist, or content manager.
- **Possible action:** Review the page and decide whether to refresh, expand, protect, prune, or continue monitoring it.
- **Cost of a false positive:** The team may waste time reviewing or changing a page that did not need attention. An unnecessary change could also harm a page that was performing well.
- **Cost of a false negative:** The team may miss an important declining page and allow it to continue losing visibility, clicks, or traffic.
- **Why data or ML may help:** A team may have thousands of pages and many signals to consider. A data-supported ranking can combine these signals and help prioritize human review more consistently. The system would support the decision; it would not make the final editing decision automatically.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd

data_url = (
    "https://raw.githubusercontent.com/"
    "Abdelrahmanshaheen1/flyrank-ml-internship/"
    "main/data/raw/content_refresh_anonymized.csv"
)

df = pd.read_csv(data_url)

total_pages = len(df)

declining_mask = df["trend_direction"].str.lower().eq("down")
declining_pages = declining_mask.sum()
declining_rate = declining_mask.mean() * 100

visible_declining_mask = declining_mask & (df["impressions_90d"] >= 500)
visible_declining_pages = visible_declining_mask.sum()

print(f"Total pages in the starter dataset: {total_pages:,}")
print(
    f"Declining pages: {declining_pages:,} "
    f"({declining_rate:.1f}% of all pages)"
)
print(
    f"Declining pages with at least 500 impressions: "
    f"{visible_declining_pages:,}"
)

Total pages in the starter dataset: 30,000
Declining pages: 16,262 (54.2% of all pages)
Declining pages with at least 500 impressions: 9,961


### Why these numbers support this lane

The starter dataset contains 30,000 webpages, which is too many for a content team to review manually. Of these pages, 16,262, or 54.2%, are labeled as declining. More importantly, 9,961 declining pages still received at least 500 impressions during the last 90 days. This suggests that many pages may still have meaningful visibility while losing performance. A ranked review queue could help the team focus first on declining pages where timely action may be most useful, instead of treating every page equally.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### What I can and cannot claim

This project may identify observed patterns in the starter data and produce a ranked list of pages that appear worth reviewing. It can support a content team by showing which pages have signals associated with decline or opportunity.

However, the project cannot prove that any single feature caused a page to decline. It cannot prove how Google's algorithm works, guarantee that editing a recommended page will improve performance, or replace human judgment. The final output should be treated as decision support, not an automatic decision.


## Self-check

Before you submit, confirm each line honestly:

- [ ✔] Every section above is filled — markdown thinking AND the code that backs it
- [ ✔] The notebook runs top to bottom with no errors (Runtime → Run all)
- [✔ ] No client names, URLs, or private queries anywhere
- [✔ ] My claims use careful words: observed, measured, directional, decision-support
- [ ✔] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.